common/shared.c

```diff
@@ -80,47 +80,62 @@ static int getaddr(char *node, char *service, void **addr,
 		return 0;
 	}
 
-	ret = getaddrinfo(node, service, NULL, &ai);
+	ret = fi_getinfo(FT_FIVERSION, node, service, flags,
+			NULL, &fi);
 	if (ret) {
-		FT_ERR("getaddrinfo error %s\n", gai_strerror(ret));
+		FT_ERR("fi_getinfo error %s\n", fi_strerror(ret));
 		return ret;
 	}
 
-	if ((*addr = malloc(ai->ai_addrlen))) {
-		memcpy(*addr, ai->ai_addr, ai->ai_addrlen);
-		*len = (size_t)ai->ai_addrlen;
+	if (flags & FI_SOURCE) {
+		if ((*addr = malloc(fi->src_addrlen))) {
+			memcpy(*addr, fi->src_addr,
+					fi->src_addrlen);
+			*len = fi->src_addrlen;
+		} else {
+			FT_ERR("addr allocation failed\n");
+			ret = EAI_MEMORY;
+		}
 	} else {
-		FT_ERR("addr allocation failed\n");
-		ret = EAI_MEMORY;
+		if ((*addr = malloc(fi->dest_addrlen))) {
+			memcpy(*addr, fi->dest_addr,
+					fi->dest_addrlen);
+			*len = fi->dest_addrlen;
+		} else {
+			FT_ERR("addr allocation failed\n");
+			ret = EAI_MEMORY;
+		}
 	}
 
-	freeaddrinfo(ai);
+	fi_freeinfo(fi);
 	return ret;
 }
 
 int ft_getsrcaddr(char *node, char *service, struct fi_info *hints)
 {
-	return getaddr(node, service, &hints->src_addr, &hints->src_addrlen);
+	return getaddr(node, service, &hints->src_addr,
+			&hints->src_addrlen, FI_SOURCE);
 }
 
 int ft_getdestaddr(char *node, char *service, struct fi_info *hints)
 {
-	return getaddr(node, service, &hints->dest_addr, &hints->dest_addrlen);
+	return getaddr(node, service, &hints->dest_addr,
+			&hints->dest_addrlen, 0);
 }
 
-int ft_read_addr_opts(char **node, char **service, struct fi_info *hints, 
+int ft_read_addr_opts(char **node, char **service, struct fi_info *hints,
 		uint64_t *flags, struct cs_opts *opts)
 {
 	int ret;
 
-	if (opts->dst_addr) {
-		ret = ft_getsrcaddr(opts->src_addr, opts->src_port, hints);
+	if (opts->dst_node) {
+		ret = ft_getsrcaddr(opts->src_node, opts->src_port, hints);
 		if (ret)
 			return ret;
-		*node = opts->dst_addr;
+		*node = opts->dst_node;
```

Comment:
This sort of renaming ends up producing a huge amount of noise in a patch and needs to be in a separate commit.